force-app/main/default/classes/plugin/EmailLoggerPlugin.cls

```diff
@@ -187,24 +191,46 @@ public without sharing class EmailLoggerPlugin implements LoggerPlugin.Triggerab
                 recipients.add(loggingQueue.Email);
             }
         }
-        
         return recipients;
     }
 
-    private static void flagLogsSent(List<Log__c> logs) {
+    private static void processEmailResults(Messaging.SendEmailResult [] results) {
+        for (Messaging.SendEmailResult result : results) {
+            if (result.isSuccess()) {
+                logsSentCounter++;
+            }
+        }
+        logsSent = logsSentCounter > 0;
+    }
+
+    private static void postProcessLogs(List<Log__c> logs) {
+        if (logsSent) {
+            setLogsSentSucess(logs);
+        } else {
+            setLogsSentFailure(logs);
+        }
+        isFirstExecution = false;
+        update logs;
+    }
+
+    private static void setLogsSentSucess(List<Log__c> logs) {
```

Comment:
The method name is not clear until you look into its code. Let's rename it to something like markLogsAsSent() and markLogsAsFailedToSend() or 
setSendStatusSucceeded() and setSendSatatusFailed()